### PR TITLE
completed POST xapp push endpoint. Changed XappEventRequest to generi…

### PR DESCRIPTION
**Pull request by joshwalong**
…c struct to be used by both Xapp POST methods

### CODE REVIEW

**Review by JCrawsh — APPROVED** [review]
LGTM

File: xumm/models/xapp_request.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package models
 
 import anyjson "github.com/xyield/xumm-go-client/pkg/json"
 
-type XappEventRequest struct {
+type XappRequest struct {
```
Comment (JCrawsh) [inline xumm/models/xapp_request.go:5]: Can we test that request body is serialised with no body when body is empty?

File: xumm/xapp/post_xapp_push_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+				"user_token": "token",
+				"subtitle": "subtitle",
+				"body": "body"
+			}`,
+			jsonResponse: `{
+				"error": {
+				  "reference": "42d58b17-ee92-419d-b8ec-15797d10c4ed",
+				  "code": 400
+				}
+			  }`,
```
Comment (JCrawsh) [inline xumm/xapp/post_xapp_push_test.go:78]: Can we also add a test with a non empty data field to see that it serialises properly.

**Review by JCrawsh — CHANGES_REQUESTED** [review]
Looks good just needs a couple of extra test cases.

File: xumm/xapp/post_xapp_push_test.go (hunk 2 above)
Comment (joshwalong) [reply xumm/xapp/post_xapp_push_test.go:87]: added test

**Review by JCrawsh — APPROVED** [review]
LGTM

**Review by bmurphy333 — APPROVED** [review]
lgtm
